core/src/toga/command.py

```diff
@@ -17,25 +20,36 @@ def __init__(
         text: str,
         *,
         parent: Group | None = None,
+        icon: str | Path | Icon | None = None,
         section: int = 0,
         order: int = 0,
+        status_item: bool = False,
```

Comment:
I did consider this; the problem is that I couldn't find an abstraction around the existing usage of Commands that made sense to me.

The current API doesn't create "menu items" and "menus"; we create and register "Commands" (as blocks of functionality that are surfaced by the app somehow). The group structure then sorts those into menus. A toolbar then becomes an alternate way of accessing that command; a command added to a toolbar is automatically registered with the app (if it isn't already registered).

So - the entry point to commands is registering them with the app. You don't register *groups* explicitly - you need to have them for organization purposes, but you don't explicitly tell your app "these are my groups"; the app discovers them by iterating over commands.

If we introduce a StatusItem class, then the'd need to have an analog of toolbar registration; That's not big problem; but what is registered against this new collection? Either:
1. StatusItem is a subclass of group. That essentially gives us what we have in this PR, except there's an extra subclass that adds nothing except a mechanism to differentiate it from a normal group... at which point, why do we need a separate registration process? 
2. StatusItem contains a group. This is a departure from the historical norm as the *group* becomes the point of registration, but you still need to register the commands, as there's no way to get from the group the commands - so you have to register all the commands, then there's a separate process to declare your status item and the group it contains. It also means you register *commands* with a toolbar, but you register a wrapper around a group with the status item... so we've lost API consistency for command-related tooling.

I came to the conclusion that (2) was a bunch of complexity and conceptual confusion for not a whole lot of gain.

That said, one advantage to (2) is that it opens an obvious door for a separate "view-based" status item base class, so popdown menus that have complex GUIs become a possibility. However, I'm not sure whether the complexity of the registration approach it requires is worth it; and I'm sure we can find some other way to accommodate them that doesn't require the conceptual gymnastics.

Reply:
Hm, I seem to recall a discussion that we had around the need of a separate `MenuItem` class some ago, for cases where `Command` does not fit well.

That being said, maybe `Command` is indeed the right abstraction for menu items in the status bar.

Another option might be to have a `StatusItem` class and register `Commands` with it instead of registering commands with the `App` and assigning them to a status item group. This would however mean a full separation between status item and app commands. According to the documentation, this already seems to be the case at least in effect, if not in code:

> Status item groups are *not* added to the main menu; instead, an icon will be added to the system tray or notification bar, and the Commands associated with the group will be displayed in a menu when the status icon is clicked.

Reply:
I think I can see how that could work; and in many regards, the implementation wouldn't be *that* different from what is here now. There's a slightly more complex path to look up and create them menu representing the group, but that's resolvable.

However, there are some interesting edge cases in the mapping between status item and menu bar. If I have 2 commands in different groups, but both registered with the same StatusItem - does that produce 2 status items or 1? Are groups on the top-level status item menus ignored? Do they convert into sections in the status menu? A toolbar puts a separator between 2 menu items in different groups, so I guess there's precedent for interpreting groups as "sections"; but the difference with toolbar is that the underlying commands still *have* their group structure in the main menu. This wouldn't be true for Status Items.